### PR TITLE
service_bus_queue: support for `max_delivery_count`

### PR DESCRIPTION
**Pull request by tkbky**
This change adds `max_delivery_count` to the service bus queue's schema so that it is configurable.

See #1977

### CODE REVIEW

File: azurerm/resource_arm_servicebus_queue_test.go [hunk 1]
```diff
@@ -551,3 +578,26 @@ resource "azurerm_servicebus_queue" "test" {
...
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    location = "${azurerm_resource_group.test.location}"
+    sku = "standard"
+}
+
+resource "azurerm_servicebus_queue" "test" {
+    name = "acctestservicebusqueue-%d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    namespace_name = "${azurerm_servicebus_namespace.test.name}"
+    maxDeliveryCount = 20
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_servicebus_queue_test.go]: can we update this to `max_delivery_count` to match the schema field?

File: azurerm/resource_arm_servicebus_queue.go [hunk 2]
```diff
@@ -117,6 +117,12 @@ func resourceArmServiceBusQueue() *schema.Resource {
 				Optional:   true,
 				Deprecated: "This field has been removed by Azure.",
 			},
+
+			"max_delivery_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  10,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_servicebus_queue.go]: out of interest where does this default value come from? (and if so, are there any validation requirements for this field?)

File: azurerm/resource_arm_servicebus_queue_test.go (hunk 1 above)
Comment (tkbky) [reply azurerm/resource_arm_servicebus_queue_test.go]: Sure.

**Comment by tkbky** [comment]
@tombuildsstuff Thanks for reviewing the PR. I've made the requested changes.

File: azurerm/resource_arm_servicebus_queue_test.go [hunk 3]
```diff
@@ -254,6 +254,33 @@ func TestAccAzureRMServiceBusQueue_isoTimeSpanAttributes(t *testing.T) {
...
+					resource.TestCheckResourceAttr(resourceName, "max_delivery_count", "10"),
+				),
+			},
+			{
+				Config: testAccAzureRMServiceBusQueue_maxDeliveryCount(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "max_delivery_count", "20"),
+				),
+			},
+		},
```
Comment (katbyte) [inline azurerm/resource_arm_servicebus_queue_test.go]: Could we add an import check step here?

```go
{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```

Comment (tkbky) [reply azurerm/resource_arm_servicebus_queue_test.go]: Sure.

**Comment by tkbky** [comment]
@katbyte It's my miss. Thanks for the catch. I've updated the PR to include the import check step, and include the max delivery count when read.

File: azurerm/resource_arm_servicebus_queue.go (hunk 2 above)
Comment (katbyte) [reply azurerm/resource_arm_servicebus_queue.go]: I was able to set it to `999999999` but it does have to be at least `1` so we could go with  `validation.IntAtleast(1)`?

Comment (tkbky) [reply azurerm/resource_arm_servicebus_queue.go]: @katbyte yes, having it to be at least `1` makes sense to me.

File: azurerm/resource_arm_servicebus_queue_test.go [hunk 4]
```diff
@@ -551,3 +583,26 @@ resource "azurerm_servicebus_queue" "test" {
 }
 `, rInt, location, rInt, rInt)
 }
+
+func testAccAzureRMServiceBusQueue_maxDeliveryCount(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/resource_arm_servicebus_queue_test.go]: Could we align all the assignments in the TF code to match the other tests?

Comment (tkbky) [reply azurerm/resource_arm_servicebus_queue_test.go]: Definitely. Sorry for not noticing the inconsistency.

**Comment by tombuildsstuff** [comment]
Tests pass:

screenshot 2018-10-10 at 14 28 31

**Pull request merged**